src/sentry/api/endpoints/organization_metrics_estimation_stats.py

```diff
@@ -0,0 +1,171 @@
+# param is the timestamp
+MetricVolumeRow = List[Union[int, CountResult]]
+
+
+@region_silo_endpoint
+class OrganizationMetricsEstimationStatsEndpoint(OrganizationEventsV2EndpointBase):
+    """Gets the estimated volume of an organization's metric events."""
+
+    def get(self, request: Request, organization: Organization) -> Response:
+
+        with sentry_sdk.start_span(
+            op="discover.metrics.endpoint", description="get_full_metrics"
+        ) as span:
+            span.set_data("organization", organization)
+
+            try:
+                # the discover stats
+                discover_stats = self.get_event_stats_data(
+                    request,
+                    organization,
+                    get_stats_generator(use_discover=True, remove_on_demand=False),
+                )
+                # the closest we have to the stats in discover that can also be queried in metrics
+                base_discover = self.get_event_stats_data(
+                    request,
+                    organization,
+                    get_stats_generator(use_discover=True, remove_on_demand=True),
+                )
+                # the closest we have to the stats in metrics
+                base_metrics = self.get_event_stats_data(
+                    request,
+                    organization,
+                    get_stats_generator(use_discover=False, remove_on_demand=True),
+                )
+
+                estimated_volume = estimate_volume(
+                    discover_stats["data"], base_discover["data"], base_metrics["data"]
+                )
+                discover_stats["data"] = estimated_volume
+
+            except ValidationError:
+                return Response(
+                    {"detail": "Comparison period is outside retention window"}, status=400
+                )
+
+        return Response(discover_stats, status=200)
+
+
+def get_stats_generator(use_discover: bool, remove_on_demand: bool):
+    """
+    Returns a get_stats function that can fetch from either metrics or discover and
+        with or without on_demand metrics.
+    """
+
+    def get_discover_stats(
+        query_columns: Sequence[str],
+        query: str,
+        params: Dict[str, str],
+        rollup: int,
+        zerofill_results: bool,
+        comparison_delta: Optional[datetime],
+    ) -> SnubaTSResult:
+        # use discover or metrics_performance depending on the dataset
+        if use_discover:
+            module = discover
+        else:
+            module = metrics_performance
+
+        if remove_on_demand:
+            query = to_standard_metrics_query(query)
+
+        return module.timeseries_query(
+            selected_columns=query_columns,
+            query=query,
+            params=params,
+            rollup=rollup,
+            referrer=Referrer.API_ORGANIZATION_METRICS_ESTIMATION_STATS.value,
+            zerofill_results=True,
+            has_metrics=True,
```

Comment:
On demand metrics needs to be enabled by passing the following flag
```suggestion
            has_metrics=True,
            on_demand_metrics_enabled=not remove_on_demand
```